backend/backend.go

```diff
@@ -64,6 +63,36 @@ func init() {
 	electrum.SetClientSoftwareVersion(Version)
 }
 
+// fixedURLWhitelist is always allowed by SystemOpen, in addition to some
+// adhoc URLs. See SystemOpen for details.
+var fixedURLWhitelist = []string{
+	// Shift Crypto owned domains.
+	"https://shiftcrypto.ch/",
+	"https://ext.shiftcrypto.ch/",
+	"https://shop.shiftcrypto.ch/",
+	"https://guides.shiftcrypto.ch/",
+	// Exchange rates.
+	"https://www.coingecko.com/",
+	"https://www.cryptocompare.com/",
+	// Block explorers.
+	"https://blockstream.info/tx/",
+	"https://blockstream.info/testnet/tx/",
+	"http://explorer.litecointools.com/tx/",
+	"https://insight.litecore.io/tx/",
+	"https://etherscan.io/tx/",
+	"https://rinkeby.etherscan.io/tx/",
+	"https://ropsten.etherscan.io/tx/",
+	// Moonpay onramp
+	"https://www.moonpay.com/",
```

Comment:
all of the ones that were an exact match before are now prefix checks. imho exact matches are slightly better, but the code is much cleaner now...

Reply:
right, only the exact match never worked for some at all. for example, we don't actually link to the moonpay.com root, ony some /privacy URL.

the whitelist purpose is to minimize some vulnerable URL injection, right. suppose an `example.org/foo` is whitelisted. if `example.org` is compromized, most likely so is its `/foo` page: not much benefit in exact match.

now, you could say but what if a web page has an XSS and a specially crafted `example.org/foo/%20<script>alert()` would pass. but then what about existing whitelisted URLs like `blockstream.info/tx/`. even with a regexp, they don't verify anything past `/tx/`. so, even with the original regexp, the same issue remains.

my point is, each URL to check should either be decomposed and compared against a domain, path and query params, or the benefits in the regexp vs prefix matching are questionable here.

yes we could change everything to match up to a single character, like in `blockstream\.info/tx/[a-fA-F0-9]{64}` and all the redirects in `ext.shiftcrypto.ch`... the list will get very long and I'm not sure how big of an advantage it would be considering a much more complicated list to maintain.

Reply:
anywa, can always change to regexp in the future! if it turns out there's actually a good reason for it.